Planner/src/Planner.java

```diff
@@ -191,5 +191,91 @@ public void actionPerformed(ActionEvent arg0) {
 		topPanel.add(selectTaskButton, BorderLayout.LINE_START);
 		plannerFrame.add(topPanel, BorderLayout.PAGE_START);
 		plannerFrame.add(listPanel, BorderLayout.LINE_START);
+		
+		//show a dialog for edit user, when we click on the user in User's list
+		listWithUsers.addListSelectionListener(new ListSelectionListener() {
+			public void valueChanged(ListSelectionEvent evtU) {
+				JTextField userName = new JTextField(25);
+			    JTextField userSurname = new JTextField(25);
+			    JPanel userPanel = new JPanel();
+			    JButton deleteUser = new JButton("Delete User");
+			    userPanel.add(new JLabel("Edit name:"));
+			    userPanel.add(userName);
+			    userPanel.add(Box.createVerticalStrut(15)); // a spacer
+			    userPanel.add(new JLabel("Edit surname:"));
+			    userPanel.add(userSurname);
+			    userPanel.add(deleteUser);
+			    userPanel.setLayout(new BoxLayout(userPanel, BoxLayout.Y_AXIS));
+				if (evtU.getValueIsAdjusting() == false) {
+					int result = JOptionPane.showConfirmDialog(null, userPanel, 
+				               "Edit the User", JOptionPane.OK_CANCEL_OPTION);
+					 if (result == JOptionPane.OK_OPTION) {
+				         System.out.println("Name: " + userName.getText());
+				         System.out.println("Surname: " + userSurname.getText());
+				         User editUser = (User)listWithUsers.getSelectedValue();
+				         editUser.setName(userName.getText());
```

Comment:
Here we shouldn't call setName and setSurname if userName.getText is empty. So we should first check that userName.getText() is not empty. The same for tasks.